### PR TITLE
BUG: an absolute path is specified for VCL_COMPILER_DETECTION_HEADER

### LINKED ISSUE #545: Error while installing ITK 5.0 RC1 on windows
I have build ITK 5.0 RC 1 on visual studio 2017 15.9.7. When I launch the command **ninja install**, I get this error:

```
-- Up-to-date: D:/dev/library/x86-64_msvc2017/itk/5.0/include/ITK-5.0/vcl_where_root_dir.h
-- Up-to-date: D:/dev/library/x86-64_msvc2017/itk/5.0/lib/itkvcl-5.0.lib
-- Up-to-date: D:/dev/library/x86-64_msvc2017/itk/5.0/include/ITK-5.0/vcl_legacy_aliases.h
CMake Error at Modules/ThirdParty/VNL/src/vxl/vcl/cmake_install.cmake:67 (file):
  file cannot create directory:
  D:/dev/library/x86-64_msvc2017/itk/5.0/include/ITK-5.0/D:/dev/library/x86-64_msvc2017/build/itk5/Modules/ThirdParty/VNL/src/vxl/vcl.
  Maybe need administrative privileges.
Call Stack (most recent call first):
  Modules/ThirdParty/VNL/src/vxl/cmake_install.cmake:37 (include)
  Modules/ThirdParty/VNL/src/cmake_install.cmake:37 (include)
  Modules/ThirdParty/VNL/cmake_install.cmake:41 (include)
  cmake_install.cmake:81 (include)

```
In the cmake_install.cmake file, I have:

``
if("x${CMAKE_INSTALL_COMPONENT}x" STREQUAL "xDevelopmentx" OR NOT CMAKE_INSTALL_COMPONENT)
  file(INSTALL DESTINATION "${CMAKE_INSTALL_PREFIX}/include/ITK-5.0/D:/dev/library/x86-64_msvc2017/build/itk5/Modules/ThirdParty/VNL/src/vxl/vcl" TYPE FILE PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ FILES "D:/dev/library/x86-64_msvc2017/build/itk5/Modules/ThirdParty/VNL/src/vxl/vcl/vcl_compiler_detection.h")
endif()
``
If I remove the part **/D:/dev/library/x86-64_msvc2017/build/itk5/Modules/ThirdParty/VNL/src/vxl/vcl** of the path (to be consistent with the other installation cmake command), the installation proceed without error.

### Versions
5.0rc1, with RTK 2.0

### Environment

Windows 10
Visual studio 2017 15.9.7
Compilation in 64 bits
Ninja

### PR DESCRIPTION
**Pull request by dzenanz**
Trying to fix #545.

### CODE REVIEW

**Comment by dzenanz** [comment]
This doesn't fix the issue, but I think that these lines are involved in causing the issue. @hjmjohnson can you take over?

File: Modules/ThirdParty/VNL/src/vxl/vcl/CMakeLists.txt [hunk 1]
```diff
@@ -88,11 +88,11 @@ write_compiler_detection_header(
 # Install platform-specific compiler detection files
 #
 if("${VXL_INSTALL_INCLUDE_DIR}" STREQUAL "include/vxl")
-  set(_compilers_destination ${VXL_INSTALL_INCLUDE_DIR}/vcl/compilers)
-  set(_complier_detection_destination ${VXL_INSTALL_INCLUDE_DIR}/vcl)
+  set(_compilers_destination vcl/compilers)
+  set(_complier_detection_destination vcl)
```
Comment (hjmjohnson) [inline Modules/ThirdParty/VNL/src/vxl/vcl/CMakeLists.txt]: @dzenanz  Please also change "_complier_" to "_compiler_" to minimize conflicts with upstream:

https://github.com/vxl/vxl/pull/601

**Comment by hjmjohnson** [comment]
@dzenanz Do you know the windows build is failing here?

File: Modules/ThirdParty/VNL/src/vxl/vcl/CMakeLists.txt (hunk 1 above)
Comment (dzenanz) [reply Modules/ThirdParty/VNL/src/vxl/vcl/CMakeLists.txt]: We can wait for upstream to fix it, then just update from upstream using the script.

**Comment by dzenanz** [comment]
This **doesn't** fix the issue. And the same issue is present in stand-alone VXL (not part of ITK).
```text
5>-- Installing: C:/Program Files/VXL/bin/vcl.dll
5>-- Up-to-date: C:/Program Files/VXL/include/vxl/vcl/vcl_legacy_aliases.h
5>CMake Error at vcl/cmake_install.cmake:87 (file):
5>  file cannot create directory: C:/Program
5>  Files/VXL/include/vxl/vcl/C:/Misc/vxl-vs2017/vcl.  Maybe need
5>  administrative privileges.
5>Call Stack (most recent call first):
5>  cmake_install.cmake:75 (include)
```

**Comment by dzenanz** [comment]
Continuing diagnosis in vxl/vxl#601.

**Comment by hjmjohnson** [comment]
@dzenanz I think this was fixed with updating of VXL master branch.  I'm closing.

**Pull request closed without merging**